### PR TITLE
timestamp comparisons

### PR DESCRIPTION
**Pull request by Goober5000**
Add methods to compare timestamps, as well as methods to calculate deltas from existing timestamps.

Follow-up to #4286.

### CODE REVIEW

File: code/io/timer.cpp [hunk 1]
```diff
@@ -289,17 +305,111 @@ int ui_timestamp_since(UI_TIMESTAMP stamp)
...
+		return 0;
+	else if (t1.value() > t2.value())
+		return 1;
+	else
+		return -1;
+}
+
+bool timestamp_in_between(TIMESTAMP stamp, TIMESTAMP before, TIMESTAMP after)
+{
+	Assertion(stamp.isValid() && !stamp.isNever(), "timestamp_in_between was called with an Invalid or Never 'stamp' timestamp!");
```
Comment (JohnAFernandez) [inline code/io/timer.cpp:380]: If there's two conditions on an assert or assertion, we really should output which statement tripped it up.

**Review by JohnAFernandez — CHANGES_REQUESTED** [review]

File: code/io/timer.cpp (hunk 1 above)
Comment (Goober5000) [reply code/io/timer.cpp:380]: Which statement?  Do you mean whether it was 'Invalid' or 'Never'?

Is that really necessary?  That would double the number of Assertions here.  There are already three for the three separate parameters.

Comment (JohnAFernandez) [reply code/io/timer.cpp:380]: What's wrong with `Assertion(stamp.isValid() && !stamp.isNever(), "timestamp_in_between was called with an %s 'stamp' timestamp!", (stamp.isNever()) ? "isNever" : "invalid");`?

Comment (Goober5000) [reply code/io/timer.cpp:380]: Seems a bit convoluted, but I can add it.

**Review by JohnAFernandez — APPROVED** [review]

File: code/io/timer.cpp (hunk 1 above)
Comment (JohnAFernandez) [reply code/io/timer.cpp]: Thanks.  Imho, we make the code clear when we can, but clarity of Assertions and error messages take priority over that.

**Pull request merged**